File: clients/java/src/main/java/io/mockly/driver/model/CallSummary.java
package io.mockly.driver.model;

import java.util.List;
import java.util.Map;

/**
 * Summary of recorded calls for a specific HTTP mock.
 */
public class CallSummary {
    private final String mockId;
    private final long count;
    private final List<CallEntry> calls;

    public CallSummary(String mockId, long count, List<CallEntry> calls) {
        this.mockId = mockId;
        this.count = count;
        this.calls = calls;
    }

    /** The mock ID these calls belong to. */
    public String getMockId() { return mockId; }

    /** Total number of times this mock was called. */
    public long getCount() { return count; }

    /** The individual recorded request entries. */
    public List<CallEntry> getCalls() { return calls; }
}

File: clients/dotnet/src/Mockly.Driver/Models/CallSummary.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Mockly.Driver.Models;

/// <summary>Summary of recorded calls for a specific HTTP mock.</summary>
public record CallSummary(
    [property: JsonPropertyName("mock_id")] string MockId,
    [property: JsonPropertyName("count")]   long   Count,
    [property: JsonPropertyName("calls")]   List<CallEntry> Calls);
